### PR TITLE
Add tccp security group id resource to fetch ingress id

### PR DESCRIPTION
**Pull request by MarcelMue**
Tested successfully on ginger.

### CODE REVIEW

File: service/controller/clusterapi/v29/controllercontext/status.go [hunk 1]
```diff
@@ -71,6 +71,7 @@ type ContextStatusTenantClusterTCCP struct {
 	IsTransitioning   bool
 	MachineDeployment v1alpha1.MachineDeployment
 	RouteTables       []*ec2.RouteTable
+	SecurityGroups    ContextStatusTenantClusterTCCPSecurityGroups
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/controllercontext/status.go]: IMO plural only when we have a list. This is a dictionary and singular reads well for the caller. 

```suggestion
	SecurityGroup    ContextStatusTenantClusterTCCPSecurityGroup
```

File: service/controller/clusterapi/v29/machine_deployment_resource_set.go [hunk 2]
```diff
@@ -170,6 +171,20 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
 		}
 	}
 
+	var tccpSecurityGroupIDResource controller.Resource
+	{
+		c := tccpsecuritygroupid.Config{
+			CMAClient:     config.CMAClient,
+			Logger:        config.Logger,
+			ToClusterFunc: newMachineDeploymentToClusterFunc(config.CMAClient),
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/machine_deployment_resource_set.go]: Why do we need the lookup function when the resource is only used within the machine deployment controller?

File: service/controller/clusterapi/v29/resource/tccpsecuritygroupid/create.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	var groups []*ec2.SecurityGroup
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding ingress security groups")
+
+		i := &ec2.DescribeSecurityGroupsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String("tag:Name"),
+					Values: []*string{
+						aws.String(key.SecurityGroupName(cr, "ingress")),
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tccpsecuritygroupid/create.go]: This needs only the cluster ID, which can be gathered using the `LabelsGetter` interface, so a pointer to the machine deployment CR is enough.

File: service/controller/clusterapi/v29/resource/tccpsecuritygroupid/create.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding ingress security group id for tenant cluster %#q", key.ClusterID(&cr)))
+
+		if len(groups) > 1 {
+			return microerror.Maskf(executionFailedError, "expected one security group, got %d", len(groups))
+		}
+
+		if len(groups) < 1 {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find ingress security group for tenant cluster %#q", key.ClusterID(&cr)))
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tccpsecuritygroupid/create.go]: As an idea, we could have the best practise of using `yet` to indicate that what we did not found should be found eventually once reconciliation sorted out all the things. WDYT?

```suggestion
			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find ingress security group for tenant cluster %#q yet", key.ClusterID(&cr)))
```

File: service/controller/clusterapi/v29/resource/tcnp/create.go [hunk 5]
```diff
@@ -301,6 +301,9 @@ func newSecurityGroups(ctx context.Context, cl v1alpha1.Cluster, md v1alpha1.Mac
 			VPC: template.ParamsMainSecurityGroupsControlPlaneVPC{
 				CIDR: cc.Status.ControlPlane.VPC.CIDR,
 			},
+			Ingress: template.ParamsMainSecurityGroupsControlPlaneIngress{
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/create.go]: Honoring the alphabet would be some really sick shit.

File: service/controller/clusterapi/v29/resource/tcnp/template/template_main_security_groups.go [hunk 6]
```diff
@@ -49,14 +49,14 @@ const TemplateMainSecurityGroups = `
         IpProtocol: tcp
         FromPort: 30011
         ToPort: 30011
-        SourceSecurityGroupId: !Ref IngressSecurityGroup
+        SourceSecurityGroupId: {{ .SecurityGroups.ControlPlane.Ingress.ID }}
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/template/template_main_security_groups.go]: Why is this template about CP when the controller context is about TC?

File: service/controller/clusterapi/v29/resource/tcnp/testdata/case-0-basic-test.golden [hunk 7]
```diff
@@ -237,14 +237,14 @@ Resources:
         IpProtocol: tcp
         FromPort: 30011
         ToPort: 30011
-        SourceSecurityGroupId: !Ref IngressSecurityGroup
+        SourceSecurityGroupId: ingressSCID
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/testdata/case-0-basic-test.golden]: What is `SC`?

File: service/controller/clusterapi/v29/machine_deployment_resource_set.go (hunk 2 above)
Comment (xh3b4sd) [reply service/controller/clusterapi/v29/machine_deployment_resource_set.go]: When you check the TCCP outputs resource or template you see that there are no security group IDs. This was about the VPC ID.

File: service/controller/clusterapi/v29/resource/tccpsecuritygroupid/create.go (hunk 4 above)
Comment (MarcelMue) [reply service/controller/clusterapi/v29/resource/tccpsecuritygroupid/create.go]: Makes sense to me. I applied this here. Different resources should be handle in separate PR as this is getting a bit big already.

**Comment by MarcelMue** [comment]
Talked to Tim - merging this one.

**Pull request merged**